rootio/wbuffer.go

```diff
@@ -141,19 +141,22 @@ func (w *WBuffer) WriteBool(v bool) {
 	if v {
 		o = 1
 	}
-	w.err = binary.Write(w.w, binary.BigEndian, o)
+	w.err = w.w.WriteByte(o)
 }
 
-func (w *WBuffer) WriteString(s string) {
+func (w *WBuffer) WriteString(v string) { //FIXME :works only on small strings 
 	if w.err != nil {
 		return
 	}
-	b1 := []byte(s)
-	l := len(s)
-	b := make([]byte, len(b1)+l)
-	b[0] = byte(l)
-	for i := range b1 {
-		b[i+1] = b1[i]
+	l := len(v)
+	b := make([]byte, l + 1)
+	if l < 255 {
+		b[0] = byte(l)
+		copy(b[1:],[]byte(v))
+	}else{
+		b[0] = 255
+		b[1] = byte(l)
```

Comment:
no, if the string's length is larger than `255`, we need more than 1 `byte` to store that length.
`ROOT` chose to encode it over 4 bytes (_ie_ a `uint32`).
that's why the reading counterpart of `WriteString` does this:
https://github.com/go-hep/hep/blob/master/rootio/rbuffer.go#L149
```go
	if u8 == 255 {
		// large string
		n = int(r.ReadU32())
	}
```

so you need to do the reverse:
- write `255` at `b[0]`
- write the actual length of the string as a `uint32` over `b[1:5]`
- write the string starting from `b[5]`.

Reply:
i tried it this way and still doesn't work tho (cf last commit) :)